src/promptflow/promptflow/_utils/thread_utils.py

```diff
@@ -39,3 +39,18 @@ def run(self):
                 for msg in msgs:
                     self._logger.log(self._level, msg)
         self.finished.set()
+
+
+# Create a thread with the context variables from the current context.
+# Our logger is context-aware, FileHandlerConcurrentWrapper need to get FileHandler from the context.
+# When we miss to set the context, log content in worker thread will not be written to the file.
+def create_thread_with_context_vars(target_function, daemon=False, target_args=(), target_kwargs={}):
```

Comment:
How about creating a subclass of threading.Thread and expose the same interface?

Reply:
Thanks! Good suggestion.